core/src/main/java/org/infinispan/interceptors/CacheLoaderInterceptor.java

```diff
@@ -93,46 +87,22 @@ protected void injectDependencies(PersistenceManager clm, EntryFactory entryFact
 
    @Override
    public Object visitApplyDeltaCommand(InvocationContext ctx, ApplyDeltaCommand command) throws Throwable {
-      if (enabled) {
-         Object key;
-         if ((key = command.getKey()) != null) {
-            loadIfNeeded(ctx, key, command);
-         }
-      }
-      return invokeNextInterceptor(ctx, command);
+      return visitDataCommand(ctx, command);
    }
 
    @Override
    public Object visitPutKeyValueCommand(InvocationContext ctx, PutKeyValueCommand command) throws Throwable {
-      if (enabled) {
-         Object key;
-         if ((key = command.getKey()) != null) {
-            loadIfNeeded(ctx, key, command);
```

Comment:
I have updated the PR, keeping only loadIfNeeded and doing the counter increments directly there.
After I've given it more thoughts, I think that it shouldn't matter what command triggered the operation - the counters should increment once we've physically touched the cache loader, so that's what the code should be doing now. The counters here are to let user see that application slowdown is there due to many loads/misses - what caused them is another issue.